twitter_like_app/src/pages/Login/Login.jsx

```diff
@@ -1,23 +1,56 @@
-import React,{Component} from "react";
-import {Grid,Paper,Avatar,Button} from '@material-ui/core';
+import React, { Component } from "react";
+import './Login.css';
+import { Grid, Paper, Avatar, TextField, Button } from '@material-ui/core';
 import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Link from '@mui/material/Link';
 import Typography from '@mui/material/Typography';
-import './Login.css';
+import { Formik, Field, Form, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+
 
 class Login extends Component {
-  
   render() {
+    const validationSchema = Yup.object().shape({
+      username: Yup.string().email('please enter valid email').required('Required'),
+      password: Yup.string().required("Required")
+    });
+    const onSubmit = (value, props) => {
+      setTimeout(() => {
+        props.resetForm()
+        props.setSubmitting(false)
+      }, 2000)
+
+    }
+    const initialValues = {
+      username: '',
+      password: '',
+      remember: 'false'
+    }
     return (
       <Grid>
-        <Paper elevation={10} className='paperStyle'>
+        <Paper elevation={10} className="paperStyle">
           <Grid align='center'>
-            <Avatar className='avatarStyle'> <LockOutlinedIcon /></Avatar>
+            <Avatar className="avatarStyle"> <LockOutlinedIcon /></Avatar>
             <h2>Login</h2>
           </Grid>
-          <Typography align='center' variant='h5'>
-            You are logged out!
+          <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
+            {(props) => (
+              <Form>
+                <Field as={TextField} margin="dense" name='username' label='Username' variant="outlined" placeholder="Enter username" fullWidth helperText={<ErrorMessage name="username" />} />
+                <Field as={TextField} margin="dense" name='password' label='Password' variant="outlined" placeholder="Enter password" fullWidth type="password" helperText={<ErrorMessage name="password" />} />
+                <FormControlLabel control={<Field as={Checkbox} name='remember' />} label="Remember me" />
+                <Field as={Button} type='submit' fullWidth variant="contained" className="btnstyle" disabled={props.isSubmitting} >{props.isSubmitting ? "Loading" : "Login"}</Field>
```

Comment:
If there are 2 or more attributes on the element consider writing in vertical.
For example:
`<Field 
  as={TextField}
  margin="dense"
  name="username"
  ...

Reply:
Also as you can see there are a lot of repeated code when you pass the same attributes to the fields. Consider moving that to the separate component.